frontends/web/src/components/accountselector/accountselector.tsx

```diff
@@ -80,21 +79,24 @@ const DropdownIndicator: FunctionComponent<DropdownIndicatorProps<TOption>> = (p
 
 export const AccountSelector = ({ title, options, selected, onChange, onProceed }: TAccountSelector) => {
   const { t } = useTranslation();
+  const [selectedAccount, setSelectedAccount] = useState<TOption>();
+
+  useEffect(() => {
+    setSelectedAccount(options.find(opt => opt.value === selected));
+  }, [options, selected]);
+
   return (
     <>
       <h1 className="title text-center">{title}</h1>
       <Select
         className={styles.select}
         classNamePrefix="react-select"
         components={{ DropdownIndicator, SingleValue: SelectSingleValue, Option: SelectOption, IndicatorSeparator: () => null }}
-        defaultValue={selected === '' ?
-          {
-            label: t('buy.info.selectLabel'),
-            value: 'choose',
-            disabled: true
-          } :
-          options.find(opt => opt.value === selected)
-        }
+        value={selected === '' ? {
+          label: t('buy.info.selectLabel'),
+          value: 'choose',
+          disabled: true
+        } : selectedAccount}
```

Comment:
what is the difference from using defaultValue vs. value here?

Reply:
To put it simply, `defaultValue` is suitable for uncontrolled components. Whereas `value` is for controlled ones. Since we're using the `selectedAccount` state now, it's a controlled component. Thus we're now utilising the `value` prop.

`selectedAccount` state is used so we can get the updated balance value once it gets fetched asyncly. This previously wasn't possible, since we're only using `options.find(opt => opt.value === selected)` for the selected value and this doesn't get updated when `options` has the balance after it was fetched.